test/Api.IntegrationTest/AdminConsole/Controllers/PoliciesControllerTests.cs

```diff
@@ -0,0 +1,174 @@
+            Assert.True(result.GetProperty("enabled").GetBoolean());
+            Assert.Equal((int)policyType, result.GetProperty("type").GetInt32());
+        }
+
+        async Task AssertPolicy()
+        {
+            var policyRepository = _factory.GetService<IPolicyRepository>();
+            var policy = await policyRepository.GetByOrganizationIdTypeAsync(_organization.Id, policyType);
+            Assert.NotNull(policy);
+
+            Assert.True(policy.Enabled);
+            Assert.Equal(policyType, policy.Type);
+            Assert.Null(policy.Data);
+            Assert.Equal(_organization.Id, policy.OrganizationId);
+        }
+    }
+
+    [Fact]
+    public async Task PutVNext_MasterPasswordPolicy_Success()
+    {
+        // Arrange
+        var policyType = PolicyType.MasterPassword;
+        var request = new SavePolicyRequest
+        {
+            Policy = new PolicyRequestModel
+            {
+                Type = policyType,
+                Enabled = true,
+                Data = new Dictionary<string, object>
+                {
+                    { "minComplexity", 10 },
+                    { "minLength", 12 },
+                    { "requireUpper", true },
+                    { "requireLower", false },
+                    { "requireNumbers", true },
+                    { "requireSpecial", false },
+                    { "enforceOnLogin", true }
+                }
+            }
+        };
+
+        // Act
+        var response = await _client.PutAsync($"/organizations/{_organization.Id}/policies/{policyType}/vnext",
+            JsonContent.Create(request));
+
+        // Assert
+        await AssertResponse();
+
+        await AssertPolicyDataForMasterPasswordPolicy();
+        return;
+
+        async Task AssertPolicyDataForMasterPasswordPolicy()
+        {
+            var policyRepository = _factory.GetService<IPolicyRepository>();
+
+            var policy = await policyRepository.GetByOrganizationIdTypeAsync(_organization.Id, policyType);
+            Assert.NotNull(policy);
+
+            Assert.True(policy.Enabled);
+            Assert.Equal(policyType, policy.Type);
+            Assert.Equal(_organization.Id, policy.OrganizationId);
+
+            Assert.NotNull(policy.Data);
+            var data = policy.GetDataModel<MasterPasswordPolicyData>();
+            Assert.Equal(request.Policy.Data["minComplexity"], data.MinComplexity);
+            Assert.Equal(request.Policy.Data["minLength"], data.MinLength);
+            Assert.Equal(request.Policy.Data["requireUpper"], data.RequireUpper);
+            Assert.Equal(request.Policy.Data["requireLower"], data.RequireLower);
+            Assert.Equal(request.Policy.Data["requireNumbers"], data.RequireNumbers);
+            Assert.Equal(request.Policy.Data["requireSpecial"], data.RequireSpecial);
+            Assert.Equal(request.Policy.Data["enforceOnLogin"], data.EnforceOnLogin);
```

Comment:
🎨 Suggestion (non-blocking): `AssertHelper.AssertPropertyEqual` can help with this.